SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/LayeredLBStrategy.cpp

```diff
@@ -21,9 +21,16 @@ void ULayeredLBStrategy::Init()
 {
 	Super::Init();
 
-	VirtualWorkerId CurrentVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID + 1;
+	UWorld* World = GetWorld();
+	Init(World);
+}
 
-	const ASpatialWorldSettings* WorldSettings = GetWorld() ? Cast<ASpatialWorldSettings>(GetWorld()->GetWorldSettings()) : nullptr;
+void ULayeredLBStrategy::Init(UWorld* World)
+{
+	Super::Init(World);
```

Comment:
I'm unclear why we need the new `Init(World*)` overload instead of the `Init()` function where we call `GetWorld()`?

Reply:
I've actually moved all the Init() logic around in another in-flight PR, so would prefer to keep the Init() as things are for now. My implementation has moved to a `SetLayers(FLayerInfo)` method which doesn't actually depend on `GetWorld()` at all so hopefully there shouldn't be any issues there 👍

Reply:
Happy to help migrate this if mine goes in first btw!

Reply:
The issue is that when I call GetWorld() from within the SpatialDebuggerEditor class it gets the world for the current map, which works. However, if the same call GetWorld() is from within the ULayeredLBStrategy the world is NULL. 

Perhaps I am not creating the load balancing strategy correctly?
 ULayeredLBStrategy* LoadBalanceStrategy = NewObject<ULayeredLBStrategy>();